src/main/java/org/skriptlang/skript/lang/structure/Structure.java

```diff
@@ -116,6 +117,9 @@ public final boolean init(Expression<?>[] expressions, int matchedPattern, Kleen
 		if (structureInfo.simple) { // simple structures do not have validators
 			return init(literals, matchedPattern, parseResult, null);
 		}
+		// make the top-level annotations visible inside this structure
+		// note: simple structures don't need this since they have no contents
+		structureData.setAnnotations(this.getParser().copyAnnotations());
```

Comment:
I'm also confused about why this is done here. What is the scenario we are considering? Is the goal to be able to access a Structure's annotations from anywhere?